### ISSUE TITLE
Rule: require tests for rules

### ISSUE THREAD

**Issue by not-an-aardvark** [not-an-aardvark/eslint-plugin-eslint-plugin#2]
Requires a rule file to have tests. Can be configured with a location of the tests.

```js
/* eslint eslint-plugin/require-rule-tests: ["error", {"testPath": "../../tests/lib/rules/"}]
```

If the file at `../../tests/lib/rules/${__filename}` exists (relative to rule file), an error is reported.

**Comment by Aladdin-ADD** [comment]
should also require docs?

**Comment by not-an-aardvark** [comment]
Seems reasonable. The reason I'm not sure about whether this rule is a good idea is because it seems like adding tests for a feature isn't necessarily something specific to ESLint plugins, so it might not be in scope for this plugin.
